packages/crypto/src/utils/base58.ts

```diff
@@ -1,22 +1,26 @@
 import { base58 } from "bstring";
+import moize from "fast-memoize";
 import { HashAlgorithms } from "../crypto";
 
-export class Base58 {
-    public static encodeCheck(buffer: Buffer): string {
-        const checksum: Buffer = HashAlgorithms.hash256(buffer);
+const encodeCheck = (buffer: Buffer): string => {
+    const checksum: Buffer = HashAlgorithms.hash256(buffer);
 
-        return base58.encode(Buffer.concat([buffer, checksum], buffer.length + 4));
-    }
-
-    public static decodeCheck(address: string): Buffer {
-        const buffer: Buffer = base58.decode(address);
-        const payload: Buffer = buffer.slice(0, -4);
-        const checksum: Buffer = HashAlgorithms.hash256(payload);
+    return base58.encode(Buffer.concat([buffer, checksum], buffer.length + 4));
+};
 
-        if (checksum.readUInt32LE(0) !== buffer.slice(-4).readUInt32LE(0)) {
-            throw new Error("Invalid checksum");
-        }
+const decodeCheck = (address: string): Buffer => {
+    const buffer: Buffer = base58.decode(address);
+    const payload: Buffer = buffer.slice(0, -4);
+    const checksum: Buffer = HashAlgorithms.hash256(payload);
 
-        return payload;
+    if (checksum.readUInt32LE(0) !== buffer.slice(-4).readUInt32LE(0)) {
+        throw new Error("Invalid checksum");
     }
-}
+
+    return payload;
+};
+
+export const Base58 = {
+    encodeCheck: moize(encodeCheck),
```

Comment:
How much data is this going to store in memory? There is no configuration or anything about that mentioned in https://www.npmjs.com/package/fast-memoize